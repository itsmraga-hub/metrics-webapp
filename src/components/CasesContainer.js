import { useState } from "react";
import { useDispatch } from "react-redux";
import { searchByCountry } from "../redux/covidcases";
import Case from "./Case";
import SearchBar from "./Search";

import style from './css/CasesContainer.module.css';


const CasesContainer = (props) => {
  const countryCases = props;
  const { cases } = countryCases;
  console.log(cases[0]);

  const dispatch = useDispatch();

  const [v, setValue] = useState({
    value: '',
  });

  const handleChange = (e) => {
    setValue({
      value: e.target.value,
    });
    const str = e.target.value;
    const searchStr = str.charAt(0).toUpperCase();
    dispatch(searchByCountry(searchStr));
  };

  return (
    <section className={style.Section}>
      <SearchBar handleChange={handleChange} value={v.value} />
      <ul className={style.Container}>
      {
        cases.map((c) => {
          const { id } = c;
          return <Case key={`key-${id}`} country={c}/>
        })
      }
    </ul>
    </section>
  )
};


export default CasesContainer;
