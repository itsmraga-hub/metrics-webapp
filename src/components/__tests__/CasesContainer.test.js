import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';

import CasesContainer from '../CasesContainer';

it('<CasesContainer /> Renders Properly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CasesContainer cases={[]} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
