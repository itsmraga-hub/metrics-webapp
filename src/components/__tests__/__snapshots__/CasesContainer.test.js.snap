// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CasesContainer /> Renders Properly 1`] = `
<section
  className="Section"
>
  <div
    className="SearchContainer"
  >
    <input
      className="InputField"
      id="searchbar"
      name="search"
      placeholder="Filter/search by country"
    />
  </div>
  <ul
    className="Container"
  />
</section>
`;
